### PR TITLE
Move current code to internal/pppoe.

### PR DESCRIPTION
**Pull request by danderson**
Also create a testutil internal package to handle all the docker
nonsense for future packages.

### CODE REVIEW

**Comment by danderson** [comment]
I should add: yes, there's no public api to internal/pppoe yet. That's the next PR.

File: internal/testutil/util.go [hunk 1]
```diff
@@ -1,5 +1,5 @@
-// package ppp implements establishing a PPP connection running over PPPoE.
-package ppp
+// package testutil contains some helpers for testing PPP and PPPoE connections.
```
Comment (bradfitz) [inline internal/testutil/util.go]: Capital P in Package

File: internal/testutil/util.go [hunk 2]
```diff
@@ -21,15 +21,16 @@ func canUseRawSockets() error {
...
 	if err != nil {
 		return fmt.Errorf("creating PPPoE Discovery listener: %v", err)
 	}
 	defer conn.Close()
 	return nil
 }
 
-func canTest() error {
+// CanRunPrivilegedTests returns nil if Docker and raw socket based tests can be run.
+func CanRunPrivilegedTests() error {
```
Comment (bradfitz) [inline internal/testutil/util.go]: "Can" sounds like it returns a bool.

I'd call this "Check" something? But it's not a huge deal, and it's a testutil.

File: internal/testutil/util.go [hunk 3]
```diff
@@ -39,7 +40,10 @@ func canTest() error {
 	return nil
 }
 
-func startServer() (func(), error) {
+// StartServer runs a PPP+PPPoE server in a Docker container. Returns
```
Comment (bradfitz) [inline internal/testutil/util.go]: "It returns ..." (complete sentence)

**Review by bradfitz — APPROVED** [review]

**Pull request merged**